crates/core/src/store.rs

```diff
@@ -442,15 +447,31 @@ impl StdinStream for MyStdinStream {
     }
 }
 
-struct MyStdoutStream(Mutex<Option<Box<dyn HostOutputStream>>>);
+#[derive(Clone)]
+struct MyStdoutStream(Arc<Mutex<Box<dyn HostOutputStream>>>);
+
+impl HostOutputStream for MyStdoutStream {
+    fn write(&mut self, bytes: Bytes) -> Result<(), StreamError> {
+        self.0.lock().unwrap().write(bytes)
+    }
+
+    fn flush(&mut self) -> Result<(), StreamError> {
+        self.0.lock().unwrap().flush()
+    }
+
+    fn check_write(&mut self) -> Result<usize, StreamError> {
+        self.0.lock().unwrap().check_write()
+    }
+}
+
+#[async_trait]
+impl Subscribe for MyStdoutStream {
+    async fn ready(&mut self) {}
```

Comment:
This should call `self.0.lock().await.ready().await` (and see the note about using `tokio::sync::Mutex` above).